Runtime/Scripts/MobiledgeXIntegrationHelper.cs

```diff
@@ -82,44 +90,50 @@ public async Task<bool> Register(string dmeHost = null, uint dmePort = 0)
             }
             catch (CarrierInfoException cie)
             {
-                Debug.LogError("Register Exception: " + cie.Message);
+                Debug.LogError("MobiledgeX: Register Exception: " + cie.Message);
                 throw new RegisterClientException(cie.Message);
             }
 
             RegisterClientReply reply = null;
             try
             {
-                if (dmeHost == null || dmePort == 0)
+                if (!useSelectedRegionInProduction)
                 {
-                    Debug.Log("Doing Register Client, with req: " + req);
+#if UNITY_EDITOR
+                        Debug.Log("MobiledgeX: Doing Register Client with DME: " + region + ", p: " + MatchingEngine.defaultDmeRestPort + " with req: " + req);
+                        Debug.LogWarning("MobiledgeX: Region Selection will work only in UnityEditor not on Mobile Devices");
+                        reply = await matchingEngine.RegisterClient(region, MatchingEngine.defaultDmeRestPort, req);
+#else
+                    Debug.Log("MobiledgeX: Doing Register Client, with req: " + req);
                     reply = await matchingEngine.RegisterClient(req);
+#endif
                 }
                 else
                 {
-                    Debug.Log("Doing Register Client with DME: " + dmeHost + ", p: " + dmePort + " with req: " + req);
-                    reply = await matchingEngine.RegisterClient(dmeHost, dmePort, req);
+                    Debug.Log("MobiledgeX: Doing Register Client with DME: " + region + ", p: " + MatchingEngine.defaultDmeRestPort + " with req: " + req);
```

Comment:
Region is not mcc-mnc, and the parameters have removed dme and port to allow SIM usage, potentially breaking some devices.

Reply:
I have (Editor Only) comment next to the region
<img width="405" alt="Screen Shot 2020-10-14 at 11 37 36 PM" src="https://user-images.githubusercontent.com/24863504/96086181-c04af600-0e76-11eb-80bf-82dd0c614709.png">